narwhal/executor/src/subscriber.rs

```diff
@@ -231,8 +233,37 @@ impl<Network: SubscriberNetwork> Fetcher<Network> {
             batches: Vec::with_capacity(num_certs),
         };
 
+        let mut workers_for_certificates: HashSet<NetworkPublicKey> = HashSet::new();
+        let mut batch_digests: HashMap<WorkerId, HashSet<BatchDigest>> = HashMap::new();
+
+        for cert in &sub_dag.certificates {
+            for (digest, (worker_id, _)) in cert.header.payload.iter() {
+                let mut workers = self.network.workers_for_certificate(cert, worker_id);
+                workers.shuffle(&mut ThreadRng::default());
+                workers_for_certificates.extend(workers.clone());
```

Comment:
```suggestion
                let workers = self.network.workers_for_certificate(cert, worker_id);
                workers_for_certificates.extend(workers);
```

Does this work as well?